src/tike/communicators/pool.py

```diff
@@ -91,26 +107,63 @@ def f(worker):
 
         return list(self.map(f, self.workers))
 
-    def scatter(self, x):
+    def scatter(self, x, s=1) -> list:
         """Split x along 0th dimension and send chunks to workers`."""
 
-        def f(worker, chunk):
-            return self._copy_to(chunk, worker)
+        def f(worker):
+            idx = worker // s
+            return self._copy_to(x[idx], worker)
 
-        return self.map(f, self.workers, x)
+        return self.map(f, self.workers)
 
-    def reduce_gpu(self, x: list, worker=None) -> cp.array:
-        """Reduce x by addition to one GPU from all other GPUs."""
-        if self.num_workers == 1:
-            return x[0]
-        worker = self.workers[0] if worker is None else worker
-        with cp.cuda.Device(worker):
-            for part in x[:worker]:
+    def scatter_bcast(self, x: list, stride=1):
+        """Send x chunks to some workers and then copy to remaining workers."""
```

Comment:
There is documentation needed for new `stride`, `s` parameters. It's not immediately obvious to me what the `scatter_bcast` function does in the context of a single host with multiple devices.